scene/gui/spin_box.cpp

```diff
@@ -41,7 +41,13 @@ Size2 SpinBox::get_minimum_size() const {
 }
 
 void SpinBox::_update_text(bool p_keep_line_edit) {
-	String value = String::num(get_value(), Math::range_step_decimals(get_step()));
+	double step = 0.0;
+	if (get_step() != 0.0 && ((get_custom_arrow_step() == 0.0) ^ (get_step() < get_custom_arrow_step()))) {
```

Comment:
```suggestion
	if (get_step() != 0.0 && ((get_custom_arrow_step() == 0.0) != (get_step() < get_custom_arrow_step()))) {
```
Clearer and more correct than bitwise operation on bools IMO

Reply:
Fixed!